packages/v4/src/content/design-guidelines/components/tree-view/tree-view.md

```diff
@@ -3,5 +3,91 @@ id: Tree view
+* **[Checkboxes](#adding-checkboxes-to-a-tree-view)** for selecting items in the tree view
+* **[Badges](#adding-badges-to-a-tree-view)** for showing the number of child nodes in the tree view
+* **[Icons](#adding-icons-to-a-tree-view)** for visually representing the node types in the tree view
+* **[Action items](#adding-action-items-to-a-tree-view)** for adding an action to each item in the tree view
+
+### Adding a search bar to a tree view
+
+Add a search bar to a tree view if you're including a large amount of data and need to provide users with a quick way to find a single item. If your dataset is small and easy to navigate, then you don't need to add a search bar.
+
+<img src="./img/with-search-bar.png" width="940" />
+
+### Adding checkboxes to a tree view
+
+Add checkboxes to a tree view if you’re selecting multiple items or applying multiple filters. Selecting should be a primary function of your use-case. If your dataset is small and you select just 1 item, use [select lists](/components/select), [radio buttons](/components/radio), [checkboxes](/components/checkbox), or a [dropdown](/components/dropdown) instead. 
+
+<img src="./img/with-checkboxes.png" width="932" />
+
+### Adding badges to a tree view
+
+Add badges to a tree view if you want to show the number of items inside. Don’t use it for displaying different information. Items without children can't have badges as they don't contain anything.
+
+<img src="./img/with-badges.png" width="332" />
+
+### Adding icons to a tree view
+
+Add icons to a tree view if you want to visually representing the type of items in a tree. For example, a folder system. If you choose to add icons to a tree view, apply the icons on all items in a tree, to maintain consistency. We don’t recommend using icons and checkboxes at the same time due to the large indentation and loss of space it would result in. 
+
+<img src="./img/with-icons.png" width="332" />
+
+### Adding action items to a tree view
+
+Add action items to a tree view if you’re triggering some type of action above the items. These actions appear on hover to avoid overcrowding the tree view. We recommend using 1 type of action across the whole tree and to limit the maximum of actions for 1 item to 1.
+
+<img src="./img/with-action-items.png" width="632" />
+
+## Tree view in a page
+
+Tree views can be included in pages within these components: [dropdown](/components/dropdown), [drawer](/components/drawer), [primary-detail](/demos/primary-detail), [modal](/components/modal), or [wizard](/components/wizard). 
+
+### Tree view in a wizard
+A tree view in a wizard allows users to apply multiple filters or select multiple items that are specific for the current step.
+
+<img src="./img/tree-view-wizard.png" width="1500" />
+
+### Tree view in a primary-detail or drawer 
+A tree view in a [primary-detail](/demos/primary-detail) or [drawer](/components/drawer) allows users to switch content based on user's selection from a tree. 
```

Comment:
I would change this a little to say "allows users to switch content based on **their** selection from a tree"